### DISCUSSION TITLE
Mongoose/MongoDB: define a custom id or use _id?

### DISCUSSION THREAD

**Discussion by Ozanyurdakul98** [neuefische/questions#290] [Web]
This is a short question about id's. Somewhere I have read that using the _id from MongoDB is not intended for use in Clientside, rather it should be used, like, only in MongoDB side. This due to security reasons.

Am I wrong here and can/should I therefore use the _id from MongoDB/mongoose to prevent a little of extra work?

**Comment by ThomasF85** [comment]
The MongoDb ObjectId that is being generated exposes some information (mostly timestamp of creation) [see here](https://www.mongodb.com/docs/v4.0/reference/method/ObjectId/). I don't see any security risk here from exposing the id, only the exposure of some information. You should apply access permissions on private objects to secure data access instead of trying to keep ObjectIds unknown to  malicious parties.
